src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -178,7 +182,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             DockerAccess access = null;
             try {
                 access = createDockerAccess(minimalApiVersion);
-                ServiceHub serviceHub = serviceHubFactory.createServiceHub(project, session, access, log, logSpecFactory);
+                fetchLimit = resolvedImages.get(0).getRunConfiguration().getFetchLimit() != null
```

Comment:
Yeah, that was a misunderstanding on my side. A global parameter is good enough.

However, our problem indeed is how we can make the lookup of a container by its name and by its image more effective. Because then we could omit the call to list() altogether. And it would be much more effective, too, especially for a large set of running containers. At the time this was implemented, there was no way with the API todo this direct lookup, but it could be that there is now an alternative API call. 

I will go hunting a bit ... ;-)